content/en/blog/2023/kwok/index.md

```diff
@@ -0,0 +1,104 @@
+---
+title: "Introducing KWOK for Istio Control Plane Testing"
+description: "Looking for a reliable way to test your Istio control plane at scale? Have you ever wondered how you could set up a cluster of thousands of nodes, simulate real nodes with a low resource footprint, and test at scale without spending a lot of money on infrastructure? Look no further! KWOK makes it easy."
+publishdate: 2023-07-20
+attribution: "Shiming Zhang (DaoCloud)"
+keywords: [Istio, kwok, testing]
+---
+
+Looking for a reliable way to test your Istio control plane at scale?
+Have you ever wondered how you could set up a cluster of thousands of nodes, simulate real nodes with a low resource footprint, and test at scale without spending a lot of money on infrastructure?
+Look no further! [KWOK] makes it easy.
+
+In this article, we'll introduce KWOK and show you how to test Istio with KWOK.
+
+## What is KWOK?
+
+KWOK stands for Kubernetes WithOut Kubelet. So far, it provides two tools:
+
+`kwok`
+: `kwok` is the cornerstone of this project, responsible for simulating the lifecycle of fake nodes, pods, and other Kubernetes API resources.
+
+`kwokctl`
+: `kwokctl` is a CLI tool designed to streamline the creation and management of clusters, with nodes simulated by `kwok`.
+
+## Why use KWOK?
+
+KWOK has several advantages:
+
+- **Lightweight**: You can simulate thousands of nodes on your laptop without significant consumption of CPU or memory resources.
+- **Fast**: You can create and delete clusters and nodes almost instantly, without waiting for boot or provisioning.
+- **Compatibility**: KWOK works with any tools or clients that are compliant with Kubernetes APIs, such as kubectl, helm, kui, etc.
+- **Portability**: KWOK has no specific hardware or software requirements. You can run it using pre-built images, once Docker/Podman/Nerdctl is installed. Alternatively, binaries are also available for all platforms and can be easily installed.
+- **Flexibility**: You can configure different node types, labels, taints, capacities, conditions, etc., and you can configure different pod behaviors, status, etc. to test different scenarios and edge cases.
+
+{{< image width="90%"
+    link="./manage-clusters.svg"
+    caption="Using kwokctl to manage simulated clusters"
+    >}}
+
+## Testing Istio with KWOK
+
+For the latest information, see [KWOK with Istio].
+
+To create a test environment for the Istio control plane with KWOK, you can follow these steps:
+
+### Set up Cluster
+
+{{< text bash >}}
+$ kwokctl create cluster --runtime kind
+{{< /text >}}
+
+### Create Node
+
+{{< text bash >}}
+$ kubectl apply -f https://kwok.sigs.k8s.io/examples/node.yaml
+{{< /text >}}
+
+### Deploy Istio
+
+{{< text bash >}}
+$ istioctl install -y
+{{< /text >}}
+
+### Migrate Controllers to Real Node
+
+We need to use [KIND] to run the controller because [KWOK] itself cannot actually run a Pod.
```

Comment:
```suggestion
We need to use [KIND](https://kind.sigs.k8s.io/) to run the controller because [KWOK](https://kwok.sigs.k8s.io/) itself cannot actually run a Pod.
```